src/compute-client/src/types/dataflows.rs

```diff
@@ -247,7 +247,13 @@ where
     }
 
     /// Like `depends_on`, but appends to an existing `BTreeSet`.
+    ///
+    /// This method includes identifiers for e.g. intermediate views, and should be filtered
+    /// if one only wants sources and indexes.
+    ///
+    /// This method is safe for mutually recursive view defintions.
     pub fn depends_on_into(&self, collection_id: GlobalId, out: &mut BTreeSet<GlobalId>) {
+        out.insert(collection_id);
```

Comment:
Yeah, great question. I think my conclusion a year ago was that the behavior isn't documented or tested in the first place, so if it passes our tests (and is now documented) that is good enough. Somewhat antisocial take on my part, but ... yeah.

Reply:
No problem at all. :-D  One immediate consequence (that is admittedly not extremely consequential):
````
materialize=> CREATE TABLE t (a int);
CREATE TABLE
materialize=> INSERT INTO t VALUES (42);
INSERT 0 1
materialize=> CREATE DEFAULT INDEX ON t;
CREATE INDEX
materialize=> CREATE VIEW v AS SELECT * FROM t;
CREATE VIEW
materialize=> CREATE DEFAULT INDEX ON v;
CREATE INDEX
materialize=> SELECT * FROM mz_internal.mz_worker_compute_dependencies;
 export_id | import_id | worker_id 
-----------+-----------+-----------
 u2        | u1        | 0
 u5        | u1        | 0
 u5        | u2        | 0
 u5        | u4        | 0
(4 rows)

materialize=> SELECT * FROM mz_internal.mz_compute_import_frontiers;
 export_id | import_id |     time      
-----------+-----------+---------------
 u2        | u1        | 1671187113468
(1 row)

materialize=> SELECT * FROM mz_internal.mz_worker_compute_delays;
 export_id | import_id | worker_id |      delay      | count 
-----------+-----------+-----------+-----------------+-------
 u2        | u1        | 0         | 00:00:00.000262 |   175
 u2        | u1        | 0         | 00:00:00.000524 |    10
 u2        | u1        | 0         | 00:00:00.001048 |     3
 u2        | u1        | 0         | 00:00:00.000032 |     9
 u2        | u1        | 0         | 00:00:00.002097 |     1
 u2        | u1        | 0         | 00:00:00.000065 |    54
 u2        | u1        | 0         | 00:00:00.000131 |   191
(7 rows)

````
So now we see the table and the view (!) as dependencies in `mz_internal.mz_worker_compute_dependencies`, but these two dependencies do not appear in `mz_internal.mz_worker_compute_delays` and `mz_internal.mz_compute_import_frontiers`. Since these are internal catalog relations, I am happy to side with "let's just be attentive when writing our debugging queries" and "mz_internal.mz_worker_compute_dependencies shows object and index dependencies, even if only the index is used for maintenance".